apps/web/src/modules/job/post/hooks/useJobForm/schema.ts

```diff
@@ -0,0 +1,70 @@
+import { Gender } from '@modela/dtos'
+import dayjs, { type Dayjs } from 'dayjs'
+import isNumeric from 'validator/lib/isNumeric'
+import { z } from 'zod'
+
+const shootingSchema = z
+  .object({
+    location: z.string({ required_error: 'กรุณากรอกสถานที่ถ่ายทำ' }),
+    startDate: z.instanceof(dayjs as unknown as typeof Dayjs),
+    endDate: z.instanceof(dayjs as unknown as typeof Dayjs),
+    startTime: z.instanceof(dayjs as unknown as typeof Dayjs),
+    endTime: z.instanceof(dayjs as unknown as typeof Dayjs),
+  })
+  .superRefine(({ startTime, endTime }, ctx) => {
+    if (startTime && endTime && startTime.isAfter(endTime)) {
+      ctx.addIssue({
+        code: 'custom',
+        path: ['endTime'],
+        message: 'เวลาสิ้นสุดการถ่ายทำต้องมากกว่าเวลาเริ่มต้น',
+      })
+    }
+  })
+  .superRefine(({ startDate, endDate }, ctx) => {
+    if (startDate && endDate && startDate.isAfter(endDate)) {
+      ctx.addIssue({
+        code: 'custom',
+        path: ['endDate'],
+        message: 'วันที่สิ้นสุดการถ่ายทำต้องมากกว่าวันที่เริ่มต้น',
+      })
+    }
+  })
+
+const postJobSchema = z
+  .object({
+    jobName: z.string().min(1, 'กรุณากรอกตำแหน่งงาน'),
```

Comment:
Required criteria will trigger on `.string` first so we need to specify message here
```suggestion
    jobName: z.string({ required_error: 'กรุณากรอกตำแหน่งงาน' }),
```

Reply:
Please apply to all relevant